src/Uno.UI/UI/Xaml/Controls/Grid/Grid.cs

```diff
@@ -1,351 +1,1805 @@
+		//  Synopsis: Sort definitions for final size processing in ArrangeOverride, for the case
+		//                 when the combined size of all definitions across a dimension exceeds the
+		//                 finalSize in that dimension.
+		//
+		//------------------------------------------------------------------------
+		void SortDefinitionsForOverflowSizeDistribution(
+			IList<DefinitionBase> ppDefinitions,
+			Xuint cDefinitions)
 		{
-			if (!_childSubscriptions.ContainsKey(child))
+			DefinitionBase pTemp;
+
+			// use insertion sort...it is stable...
+			for (Xuint i = 1, j; i < cDefinitions; i++)
 			{
-				// The tradeoff made here is to assume that registering to all property changes, and checking for DP
-				// equality in the callback is less expensive that forcing the initialization of the four grid attached properties 
-				// storage on all the children.
-				_childSubscriptions[child] = 
-					DependencyObjectExtensions
-						.RegisterDisposablePropertyChangedCallback(child, OnChildGridPropertyChanged);
+				pTemp = ppDefinitions[i];
+				for (j = i; j > 0; j--)
+				{
+					if ((pTemp.GetMeasureArrangeSize() - pTemp.GetEffectiveMinSize())
+					    >= (ppDefinitions[j - 1].GetMeasureArrangeSize() - ppDefinitions[j - 1].GetEffectiveMinSize()))
+					{
+						break;
+					}
+
+					ppDefinitions[j] = ppDefinitions[j - 1];
+				}
+
+				ppDefinitions[j] = pTemp;
 			}
 		}
 
-		/// <summary>
-		/// Removes the attached properties registrations for the specified object.
-		/// </summary>
-		/// <param name="view"></param>
-		private void DisposeGridChildPropertiesSubscription(NativeView view)
+
+		//------------------------------------------------------------------------
+		//
+		//  Method: SortDefinitionsForStarSizeDistribution
+		//
+		//  Synopsis: Sort definitions for distributing star space.
+		//
+		//------------------------------------------------------------------------
+		void SortDefinitionsForStarSizeDistribution(
+				IList<DefinitionBase> ppDefinitions,
+		Xuint cDefinitions
+		)
 		{
-			IDisposable d;
+			DefinitionBase pTemp;
 
-			if (_childSubscriptions.TryGetValue(view, out d))
+			// use insertion sort...it is stable...
+			for (Xuint i = 1, j; i < cDefinitions; i++)
 			{
-				d.Dispose();
-				_childSubscriptions.Remove(view);
+				pTemp = ppDefinitions[i];
+				for (j = i; j > 0; j--)
+				{
+					// Use >= instead of > to keep sort stable. If > is used,
+					// sort will not be stable & size will distributed in a different
+					// order than WPF.
+					if (pTemp.GetSizeCache() >= ppDefinitions[j - 1].GetSizeCache())
+					{
+						break;
+					}
+
+					ppDefinitions[j] = ppDefinitions[j - 1];
+				}
+
+				ppDefinitions[j] = pTemp;
 			}
 		}
+
+		//void OnPropertyChanged(PropertyChangedParams args)
+		//{
+		//	//Panel.OnPropertyChanged(args);
+
+		//	if (args.Property == Grid.RowDefinitionsProperty
+		//	    || args.Property == Grid.ColumnDefinitionsProperty)
+		//	{
+		//		InvalidateDefinitions();
```

Comment:
Implemented somewhere else?

Reply:
Yep.  It's in the `OnGenericPropertyChanged()` of the `Grid.Properties.cs` file.